internal/provider/models/api_token.go

```diff
@@ -67,3 +86,48 @@ func (data *ApiTokenDataSource) ReadFromResponse(ctx context.Context, apiToken *
 	}
 	return diags
 }
+
+func (data *ApiTokenResource) ReadFromResponse(ctx context.Context, apiToken *iam.ApiToken) diag.Diagnostics {
+	var diags diag.Diagnostics
+	data.Id = types.StringValue(apiToken.Id)
+	data.Name = types.StringValue(apiToken.Name)
+	if apiToken.Description != "" {
+		data.Description = types.StringValue(apiToken.Description)
+	}
```

Comment:
I feel like this would make more sense
```suggestion
	data.Description = types.StringValue(apiToken.Description)
```

Reply:
or maybe this since description is `optional`
```suggestion
	if apiToken.Description == "" {
		data.Description = types.StringNull()
	} else {
	        data.Description = types.StringValue(apiToken.Description)
	}
```

regardless we should add test cases where we create a token with empty description then update it to non-empty and vice versa case

Reply:
tested it with the second code snippet and it works fine, but it also works fine without explicitly setting the value to null